src-java/network-topology/network-storm-topology/src/main/java/org/openkilda/wfm/topology/network/storm/bolt/isl/IslHandler.java

```diff
@@ -219,17 +220,18 @@ public void islRemovedNotification(Endpoint srcEndpoint, IslReference reference)
     public void islChangedNotifyFlowMonitor(IslReference reference) {
         Optional<Endpoint> src = Optional.ofNullable(reference.getSource());
         Optional<Endpoint> dst = Optional.ofNullable(reference.getDest());
-        IslChangedInfoData islChangedInfoData = IslChangedInfoData.builder()
+        IslChangedInfoDataBuilder islChangedInfoData = IslChangedInfoData.builder()
                 .source(NetworkEndpoint.builder()
                         .datapath(src.map(Endpoint::getDatapath).orElse(null))
                         .portNumber(src.map(Endpoint::getPortNumber).orElse(null))
-                        .build())
-                .destination(NetworkEndpoint.builder()
-                        .datapath(dst.map(Endpoint::getDatapath).orElse(null))
-                        .portNumber(dst.map(Endpoint::getPortNumber).orElse(null))
-                        .build())
-                .build();
-        emit(STREAM_FLOW_MONITORING_ID, getCurrentTuple(), makeIslChangedTuple(islChangedInfoData));
+                        .build());
+        if (dst.isPresent()) {
+            islChangedInfoData.destination(NetworkEndpoint.builder()
+                    .datapath(dst.map(Endpoint::getDatapath).orElse(null))
+                    .portNumber(dst.map(Endpoint::getPortNumber).orElse(null))
```

Comment:
we need more lambdas... :(

Reply:
Reworked.

Reply:
why not just `if (reference.getDest() == null) { doSomething(); }`? Just curious... do not understand such lambdas popularity in java.

Reply:
As for me lambdas and other functional-style code in java is more human-readable so I like to use them when I can.